### PR TITLE
added sonarqube analysis and github action workflow integration

### PR DESCRIPTION
**Pull request by arshid75**
## What does this PR do?

runs sonar analysis on pull requests and on push to master and dev branches.

## Changes

- added sonar.propeties files
- added sonar.yaml file for GitHub actions workflow.

## Before reviewing

- [ ] Are there any linked PRs?
- [x] Did you test this PR?
- [ ] Is there any migration or seeding necessary?
- [ ] Did you add api to only one of the language (python/typescript)?

### Links:

-   xyme-backend: https://github.com/Accern/xyme-backend/pull/279
-  sonar-analysis: https://sonarqube.production-v3.accern.com/dashboard?id=Accern_accern-xyme

### How did you test this PR?
checked sonar analysis in Sonarqube UI when I create new PR

### New API added that not implemented in other languages

If your answer is yes, please also update the todo_api.txt at where the API is missing.

### CODE REVIEW

File: .github/workflows/sonar.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                  key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+
+            - name: Install dependencies
+              run: yarn
+
+            - name: Type Check
+              run: yarn run typecheck
+
+            - name: Lint
+              run: yarn run lint
```
Comment (mahadeveaswar) [inline .github/workflows/sonar.yaml:48]: are the above steps like yarn checks required for sonarqube analysis?

File: .github/workflows/sonar.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+              
+            - name: SonarQube
+              uses: sonarsource/sonarqube-scan-action@master
+              env:
+                SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+                SONAR_HOST_URL: ${{ secrets.SONAR_HOST_URL }}
```
[the remark this answers is not in the document]
Comment (arshid75) [reply .github/workflows/sonar.yaml:54]: ok missed again. will add that

File: .github/workflows/sonar.yaml (hunk 1 above)
Comment (arshid75) [reply .github/workflows/sonar.yaml:48]: since we don't have any tests here, we can do this test here

File: .github/workflows/sonar.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
+name: sonar-analysis
+
+on:
+  push:
+    branches:
+      - master
+      - dev
```
Comment (JosuaKrause) [inline .github/workflows/sonar.yaml:7]: and tags

File: .github/workflows/sonar.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                  key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+                  restore-keys: |
+                      ${{ runner.os }}-yarn-
+
+            - name: Cache node modules
+              uses: actions/cache@v2
+              with:
+                  path: '**/node_modules'
+                  key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+
```
Comment (JosuaKrause) [inline .github/workflows/sonar.yaml:40]: do we have any python tests to run?

Comment (TheAnkurGoswami) [reply .github/workflows/sonar.yaml:40]: w/o that we don't have coverage

Comment (TheAnkurGoswami) [reply .github/workflows/sonar.yaml:40]: we planned of having those tests, but it's blocked by that redis test issue.

File: .github/workflows/sonar.yaml (hunk 3 above)
Comment (arshid75) [reply .github/workflows/sonar.yaml:7]: @JosuaKrause  in jira ticket, definition of done only branches and prs are include. no tags. I will remove dev then.

File: sonar-project.properties [hunk 5]
```diff
@@ -0,0 +1,4 @@
+sonar.projectKey=Accern_accern-xyme
+sonar.projectName=accern-xyme
+sonar.projectVersion=4.1.3
+sonar.sources=./packages/,./stubs/
```
[the remark this answers is not in the document]
Comment (arshid75) [reply sonar-project.properties:4]: @JosuaKrause  do you want only packages folder to be analysed?

[the remark this answers is not in the document]
Comment (arshid75) [reply sonar-project.properties:4]: @TheAnkurGoswami  @mahadeveaswar  Do you guys want stubs folder to be ignored for sonar analysis?

File: .github/workflows/sonar.yaml (hunk 3 above)
Comment (arshid75) [reply .github/workflows/sonar.yaml:7]: @mahadeveaswar  @TheAnkurGoswami  which branches other than master do you want to include in sonar analysis?

Comment (mahadeveaswar) [reply .github/workflows/sonar.yaml:7]: Just master branch is fine here and include tags

File: sonar-project.properties (hunk 5 above)
[the remark this answers is not in the document]
Comment (arshid75) [reply sonar-project.properties:4]: removing that

**Review by JosuaKrause — APPROVED** [review]
lgtm

**Pull request merged**